packages/gatsby-theme-newrelic/src/hooks/useInstrumentedHandler.js

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useEffect, useRef } from 'react';
+import warning from 'warning';
+
+const useInstrumentedHandler = (handler, attributes) => {
```

Comment:
This doesn't need to be added in this PR, but I think this would be great to expand to handle all instrumentation. For example, we could have something like: 
```js
const useInstrumentedHandler = (handler, attributes, config) => {
```
where config could be an object with defaults `{newrelic: true, tessen: false, googleAnalytics: false}. And then we would have something like: 
```js
const instrumentationConfig =  {newrelic: false, tessen: true, googleAnalytics: true};

useInstrumentedHandler(() => console.log('clicked'), { actionName: 'click' }, instrumentationConfig);
```
Therefore, for all the tickets like #254 where we want to instrument to multiple tools we could just have one easy handler.

What do you think?

Reply:
Oooo that's a really good idea.

I think the trick here is figuring out how best to unify the attributes. NR Browser requires an `actionName`, Tessen requires both a `name` and `category` (which it then combines into a single string when it sends to New Relic), and Google Analytics [requires](https://developers.google.com/analytics/devguides/collection/analyticsjs/command-queue-reference#send) `eventCategory`, `eventAction`, `eventLabel`, and `eventValue`. 

If we can figure out how to best make sure that all of the required fields for each tracker are filled out, then I really like this idea.